apps/gdalalg_raster_create.cpp

```diff
@@ -0,0 +1,453 @@
+        }
+
+        if (m_nodata.empty() && m_bandCount > 0)
+        {
+            int bNoData = false;
+            const double dfNoData =
+                poSrcDS->GetRasterBand(1)->GetNoDataValue(&bNoData);
+            if (bNoData)
+                m_nodata = CPLSPrintf("%.17g", dfNoData);
+        }
+    }
+
+    if (m_size.empty())
+    {
+        ReportError(CE_Failure, CPLE_IllegalArg,
+                    "Argument 'size' should be specified, or 'like' dataset "
+                    "should be specified");
+        return false;
+    }
+
+    if (!m_burnValues.empty() && m_burnValues.size() != 1 &&
+        static_cast<int>(m_burnValues.size()) != m_bandCount)
+    {
+        if (m_bandCount == 1)
+        {
+            ReportError(CE_Failure, CPLE_IllegalArg,
+                        "One value should be provided for argument "
+                        "'burn', given there is one band");
+        }
+        else
+        {
+            ReportError(CE_Failure, CPLE_IllegalArg,
+                        "One or %d values should be provided for argument "
+                        "'burn', given there are %d bands",
+                        m_bandCount, m_bandCount);
+        }
+        return false;
+    }
+
+    auto poDriver =
+        GetGDALDriverManager()->GetDriverByName(m_outputFormat.c_str());
+    if (!poDriver)
+    {
+        // shouldn't happen given checks done in GDALAlgorithm
+        ReportError(CE_Failure, CPLE_AppDefined, "Cannot find driver %s",
+                    m_outputFormat.c_str());
+        return false;
+    }
+
+    if (m_append)
+    {
+        if (poDriver->GetMetadataItem(GDAL_DCAP_CREATE_SUBDATASETS) == nullptr)
+        {
+            ReportError(CE_Failure, CPLE_NotSupported,
+                        "-append option not supported for driver %s",
+                        poDriver->GetDescription());
+            return false;
+        }
+        m_creationOptions.push_back("APPEND_SUBDATASET=YES");
+    }
+
+    auto poRetDS = std::unique_ptr<GDALDataset>(poDriver->Create(
+        m_outputDataset.GetName().c_str(), m_size[0], m_size[1], m_bandCount,
+        GDALGetDataTypeByName(m_type.c_str()),
+        CPLStringList(m_creationOptions).List()));
+    if (!poRetDS)
+    {
+        return false;
+    }
+
+    if (!m_crs.empty() && m_crs != "none" && m_crs != "null")
+    {
+        oSRS.SetFromUserInput(m_crs.c_str());
+        oSRS.SetAxisMappingStrategy(OAMS_TRADITIONAL_GIS_ORDER);
+    }
+
+    if (!oSRS.IsEmpty())
+    {
+        if (poRetDS->SetSpatialRef(&oSRS) != CE_None)
+        {
+            ReportError(CE_Failure, CPLE_AppDefined, "Setting CRS failed");
+            return false;
+        }
+    }
+
+    if (!m_bbox.empty())
+    {
+        if (poRetDS->GetRasterXSize() == 0 || poRetDS->GetRasterYSize() == 0)
+        {
+            ReportError(CE_Failure, CPLE_AppDefined,
+                        "Cannot set extent because dataset has 0x0 dimension");
```

Comment:
```suggestion
                        "Cannot set extent because one of dataset height or width is null"
``` 
actually, this branch is executed even if one of the two dimensions is not 0.

Reply:
fixed